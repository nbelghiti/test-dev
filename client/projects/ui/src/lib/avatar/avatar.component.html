<span
	class="c-avatar"
	[ngClass]="{
		'is-compact': compact
	}"
>
	<abbr [title]="name">{{initials}}</abbr>
</span>
