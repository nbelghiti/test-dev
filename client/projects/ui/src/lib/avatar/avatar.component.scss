// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Settings
// =============================================================================
$avatar-size-base: 4.4rem;
$avatar-size-compact: 3.2rem;

// =============================================================================
// :: Avatar
// =============================================================================
.c-avatar {
	display: flex;
	justify-content: center;
	align-items: center;
	width: $avatar-size-base;
	min-width: $avatar-size-base;
	height: $avatar-size-base;

	font-size: $font-size-1x-large;
	font-weight: $font-weight-bold;
	line-height: $line-height-base;
	text-transform: uppercase;
	text-align: center;

	border-radius: 50%;
	background-color: $background-color-grey-96;

	cursor: default;
}

// =============================================================================
// :: Modifiers
// =============================================================================
.c-avatar {

	&.is-compact {
		width: $avatar-size-compact;
		min-width: $avatar-size-compact;
		height: $avatar-size-compact;

		font-size: $font-size-base;
	}
}
