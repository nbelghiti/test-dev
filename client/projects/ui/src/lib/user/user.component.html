<ng-template #userContent>
	<ui-avatar
		[name]="user?.name"
		[compact]="true"
	>
	</ui-avatar>

	<div class="c-user__content">
		<span class="c-user__name">
			{{user?.name}}<span *ngIf="user?.myself"> (you)</span>
		</span>

		<span
				class="c-user__role"
			>
				{{ label | translate }}
			</span>
	</div>
</ng-template>

<div
	*ngIf="!isButton; else isButton"
	class="c-user"
	[ngClass]="{ 'c-user--with-background': backgroundColor }"
	[title]="user?.name"
>
	<ng-container [ngTemplateOutlet]="userContent"></ng-container>
</div>

<ng-template #isButton>
	<button
		class="c-user"
		[ngClass]="{ 'c-user--with-background': backgroundColor }"
		[title]="user?.name"
		(click)="clicked.emit(user)"
	>
		<ng-container [ngTemplateOutlet]="userContent"></ng-container>
	</button>
<ng-template>
