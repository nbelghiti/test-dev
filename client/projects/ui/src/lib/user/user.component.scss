// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: User
// =============================================================================
.c-user {
	display: flex;
	align-items: center;
	width: 100%;

	font-weight: $font-weight-bold;
	font-size: $font-size-1x-large;
	line-height: $line-height-base;

	&:hover {
		cursor: default;
	}
}

// =============================================================================
// :: Elements
// =============================================================================
.c-user__content {
	display: flex;
	flex-direction: column;
	overflow: hidden;
	margin-left: 0.6rem;

	font-size: $font-size-2x-small;
	line-height: $line-height-base;
	text-align: left;
	white-space: nowrap;

	color: $text-color-grey-46;
}

.c-user__name {
	@include extend-text-overflow();

	font-weight: $font-weight-bold;
}

.c-user__role {
	@include extend-text-overflow();

	font-weight: $font-weight-normal;
}

// =============================================================================
// :: Modifiers
// =============================================================================
.c-user--with-background {
	padding: 0.5rem 0.8rem;

	background-color: $background-color-grey-96;
}
