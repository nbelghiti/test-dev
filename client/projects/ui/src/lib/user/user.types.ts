export interface IUser {
	name?: string;
	role?: string;
	type?: string;
	myself?: boolean;
}
