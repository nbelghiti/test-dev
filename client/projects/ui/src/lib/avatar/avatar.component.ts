import { Component, Input } from '@angular/core';

import { getInitials } from './avatar.utils';

@Component({
	selector: 'ui-avatar',
	templateUrl: './avatar.component.html',
	styleUrls: ['./avatar.component.scss']
})

export class AvatarComponent {
	@Input() public compact: boolean = false;

	public initials = '';

	@Input() set name(name: string) {
		if (name) {
			this.initials = getInitials(name);
		}
	}
}
