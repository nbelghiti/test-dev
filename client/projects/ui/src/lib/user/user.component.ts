import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
import { TranslateService } from '@ngx-translate/core';

import { IUser } from './user.types';

@Component({
	selector: 'ui-user',
	templateUrl: './user.component.html',
	styleUrls: ['./user.component.scss']
})

export class UserComponent implements OnInit {
	@Input() public user: IUser;
	@Input() public backgroundColor: boolean = false;

	@Output() public clicked: EventEmitter<IUser> = new EventEmitter<IUser>();

	public isButton = false;
	public label: string = '';

	constructor(
		private translate: TranslateService
	) {}

	public ngOnInit(): void {
		const role = this.user && (this.user.role || this.user.type);
		if (role) {
			this.label = this.translate.instant(`TEAM_MEMBER_ROLE__${role}`);
		}
		this.isButton = this.clicked.observers.length > 0;
	}
}
