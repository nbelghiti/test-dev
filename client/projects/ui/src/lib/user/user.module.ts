import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { TranslateModule } from '@ngx-translate/core';

import { AvatarModule } from '../avatar/avatar.module';

import { UserComponent } from './user.component';

@NgModule({
	imports: [
		TranslateModule,
		CommonModule,
		AvatarModule,
	],
	declarations: [
		UserComponent,
	],
	exports: [
		UserComponent,
	],
})
export class UserModule {}
