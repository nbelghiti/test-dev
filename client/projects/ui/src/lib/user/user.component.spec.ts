import { ComponentFixture, TestBed, async } from '@angular/core/testing';
import { TranslateModule, TranslateService } from '@ngx-translate/core';

import { AvatarComponent } from '../avatar/avatar.component';

import { UserComponent } from './user.component';

const mockUser = {
	name: 'John',
	role: '',
	type: 'engineer',
	myself: false,
};

describe('UserComponent', () => {
	let component: UserComponent;
	let fixture: ComponentFixture<UserComponent>;
	let translate: TranslateService;

	beforeEach(async(() => {
		TestBed.configureTestingModule({
			imports: [
				TranslateModule.forRoot(),
			],
			declarations: [
				AvatarComponent,
				UserComponent
			],
			providers: [
				TranslateService,
			]
		})
			.compileComponents();

		// Get the providers from the TestBed
		translate = TestBed.get(TranslateService);
	}));

	beforeEach(() => {
		fixture = TestBed.createComponent(UserComponent);
		component = fixture.componentInstance;
		component.user = mockUser;

		fixture.detectChanges();
	});

	it('should create', () => {
		expect(component).toBeTruthy();
	});

	it('Should render correctly', () => {
		expect(fixture).toMatchSnapshot();
	});
});
