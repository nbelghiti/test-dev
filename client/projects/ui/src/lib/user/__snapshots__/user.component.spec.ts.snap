// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserComponent Should render correctly 1`] = `
<ui-user
  backgroundColor="false"
  clicked={[Function EventEmitter]}
  isButton="false"
  label={[Function String]}
  translate={[Function TranslateService]}
  user={[Function Object]}
>
  <button
    class="c-user"
    ng-reflect-klass="c-user"
    ng-reflect-ng-class="[object Object]"
    title="John"
  >
    
    <ui-avatar
      ng-reflect-compact="true"
      ng-reflect-name="John"
    >
      <span
        class="c-avatar is-compact"
        ng-reflect-klass="c-avatar"
        ng-reflect-ng-class="[object Object]"
      >
        <abbr
          title="undefined"
        >
          J
        </abbr>
      </span>
    </ui-avatar>
    <div
      class="c-user__content"
    >
      <span
        class="c-user__name"
      >
         John
        
      </span>
      <span
        class="c-user__role"
      >
         TEAM_MEMBER_ROLE__engineer 
      </span>
    </div>
  </button>
</ui-user>
`;
